src/generators/default/number/number.test.ts

```diff
@@ -14,50 +18,85 @@ describe('create numbers', () => {
-	});
+		test('creates a number with a min value', () => {
+			expect(core.generate(z.number().min(10_000))).toBeGreaterThanOrEqual(
+				10_000
+			);
+			expect(core.generate(z.number().gte(10_000))).toBeGreaterThanOrEqual(
+				10_000
+			);
+		});
 
-	test('creates a number with a negative max value', () => {
-		expect(core.generate(z.number().max(-5))).toBeLessThanOrEqual(-5);
-	});
+		test('creates a number with a negative min value', () => {
+			expect(core.generate(z.number().min(-10))).toBeGreaterThanOrEqual(-10);
+		});
 
-	test('creates a number with a negative max value', () => {
-		const result = core.generate(z.number().min(-10).max(-5));
-		expect(result).toBeGreaterThanOrEqual(-10);
-		expect(result).toBeLessThanOrEqual(-5);
-	});
+		test('creates a number with a max value', () => {
+			expect(core.generate(z.number().max(5))).toBeLessThanOrEqual(5);
+			expect(core.generate(z.number().lte(5))).toBeLessThanOrEqual(5);
+		});
+
+		test('creates a number with a negative max value', () => {
+			expect(core.generate(z.number().max(-5))).toBeLessThanOrEqual(-5);
+		});
 
-	test('creates a positive number', () => {
-		expect(core.generate(z.number().positive())).toBeGreaterThan(0);
-		expect(core.generate(z.number().nonnegative())).toBeGreaterThanOrEqual(0);
+		test('creates a number with a negative max value', () => {
+			const result = core.generate(z.number().min(-10).max(-5));
+			expect(result).toBeGreaterThanOrEqual(-10);
+			expect(result).toBeLessThanOrEqual(-5);
+		});
+
+		test('throws when min is greater than max', () => {
+			expect(() => core.generate(z.number().min(100).max(10))).toThrowError();
+		});
 	});
 
-	test('creates a negative number', () => {
-		expect(core.generate(z.number().negative())).toBeLessThan(0);
-		expect(core.generate(z.number().nonpositive())).toBeLessThanOrEqual(0);
+	describe('positive and negative', () => {
+		test('produces a valid positive number', () => {
+			expect(core).toProduce(z.number().positive());
+		});
+
+		test('creates a positive number', () => {
+			expect(core.generate(z.number().positive())).toBeGreaterThan(0);
+			expect(core.generate(z.number().nonnegative())).toBeGreaterThanOrEqual(0);
+		});
+
+		test('produces a valid negative number', () => {
+			expect(core).toProduce(z.number().positive());
+		});
+
+		test('creates a negative number', () => {
+			expect(core.generate(z.number().negative())).toBeLessThan(0);
+			expect(core.generate(z.number().nonpositive())).toBeLessThanOrEqual(0);
+		});
 	});
 
-	test('throws when min is greater than max', () => {
-		expect(() => core.generate(z.number().min(100).max(10))).toThrowError();
+	describe('multipleOf', () => {
+		test('produces a valid multipleOf number', () => {
+			expect(core).toProduce(z.number().multipleOf(33));
+		});
+
+		test("creates a number that's a multiple with min and max", () => {
+			const schema = z.number().multipleOf(3).min(16).max(20);
+			expect(core).toProduce(schema, 18);
```

Comment:
Updated these tests to use `toProduce`.
I also included the expected result to `toProduce` because for these cases it was useful.